frontend/app/utils/route-utils.ts

```diff
@@ -100,3 +100,39 @@ export function getRouteByPath(pathname: string, routes: I18nRoute[]): I18nPageR
 function normalizePath(pathname: string): string {
   return pathname.replace(/\/+$/, '');
 }
+
+/**
+ * Extracts and returns the locale from the provided request URL.
+ *
+ * This function analyzes the pathname of the given `request` URL to determine the locale. It supports
+ * English (`'en'`) and French (`'fr'`). If the pathname does not start with `/en` or `/fr`, the function
+ * defaults to English (`'en'`).
+ *
+ * @param request - The HTTP request object containing the URL from which to extract the locale.
+ * @returns The detected locale, either `'en'` or `'fr'`. Defaults to `'en'` if no valid locale is found.
+ */
+export function getLocale(request: Request): 'en' | 'fr' {
+  const { pathname } = new URL(request.url);
+  if (pathname.startsWith('/en')) {
+    return 'en';
+  }
+  if (pathname.startsWith('/fr')) {
+    return 'fr';
+  }
+  return 'en';
+}
+
+/**
+ * Recursively searches for a path matching the given file within a structure of I18nRoutes.
+ *
+ * @param i18nRouteFile - The file name of the route to search for.
+ * @param routes - The array of I18nRoutes to search within.
+ * @param request - The HTTP request object containing the URL from which to extract the locale.
+ * @returns The path of the route that matches the given file name.
+ */
+export function getPathByFile(i18nRouteFile: I18nRouteFile, routes: I18nRoute[], request: Request): string {
```

Comment:
Do we need another `getXxxByFile()` function when to get the paths all we need to do is `getRouteByFile().paths`?

Reply:
We can create our own `redirect` method for the app `appRedirect` that would accept `i18nRouteFile: I18nRouteFile` and `language: Language`. 🤷

Reply:
...or `appRedirect('path/to/file.tsx', request)`, even

Reply:
one or the other